src/jdk.incubator.foreign/share/classes/jdk/internal/foreign/abi/Binding.java

```diff
@@ -281,22 +283,47 @@ private static void checkType(Class<?> type) {
             throw new IllegalArgumentException("Illegal type: " + type);
     }
 
-    private static MemorySegment copyBuffer(MemorySegment operand, long size, long alignment, NativeScope allocator) {
+    private static void checkOffset(long offset) {
+        if (offset < 0)
+            throw new IllegalArgumentException("Negative offset: " + offset);
+    }
+
+    private static MemorySegment copyBuffer(MemorySegment operand, long size, long alignment,
+                                            SharedUtils.Allocator allocator) {
         MemorySegment copy = allocator.allocate(size, alignment);
         copy.copyFrom(operand.asSlice(0, size));
         return copy;
     }
 
-    public static Move move(VMStorage storage, Class<?> type) {
+    private static MemorySegment allocateBuffer(long size, long allignment, SharedUtils.Allocator allocator) {
+        return allocator.allocate(size, allignment);
+    }
+
+    // FIXME should register with scope
+    private static MemorySegment toSegment(MemoryAddress operand, long size) {
```

Comment:
This seems to be only used inside the `ToSegment` binding, so, perhaps this could should be moved there (also because this is an interpreter function, not a binding factory - e.g. the overload with the `toSegment` binding factory is confusing)

Reply:
Yeah, good idea. This and 2 others are used for the `specialize` implemantion, but they can be moved inside the binding classes.